packages/plugins/type-designer/src/stores/data-types-templates.store.ts

```diff
@@ -1,43 +1,161 @@
+	const newRootElement = get(dataTypeTemplateService).findDataTypesElement()
 	rootElement.set(newRootElement)
 }
 
-function setSubElements(newXmlDocument: Element) {
-	const typeDesigner = new DataTypeTemplatesService(newXmlDocument)
+function setPrivateElement() {
+	const currentRootElement = get(rootElement)
+	if (!currentRootElement) return privateElement.set(null)
+
+	const newPrivateElement = get(dataTypeTemplateService).findPrivateElement(
+		'[type="type-designer-plugin"]',
+		{
+			root: currentRootElement.element
+		}
+	)
 
-	const newSubElements = typeDesigner.findTypeDesignerElements({
-		root: get(rootElement)?.element
+	privateElement.set(newPrivateElement)
+}
+
+function setTypeElements() {
+	const currentPrivateElement = get(privateElement)
+
+	const newSubElements = get(dataTypeTemplateService).findTypeElements({
+		root: currentPrivateElement?.element
 	})
 
-	subElements.set(newSubElements)
+	typeElements.set(newSubElements)
+}
+
+function createDataTypeTemplateElement() {
+	const currentXmlDocumentElement = get(
+		pluginStore.xmlDocument
+	)?.documentElement
+	if (!currentXmlDocumentElement)
+		return console.error('No XML document found')
+
+	const newDataTypeTemplatesElement = pluginStore.addElementToXmlDocument({
+		parentElement: currentXmlDocumentElement,
+		newElementTagName: 'DataTypeTemplates'
+	})
+	rootElement.set({
+		element: newDataTypeTemplatesElement
+	})
 }
 
-//==== INITIALIZATION
-function init(newXmlDocument: Element | undefined) {
+function createPrivateElement() {
+	const currentRootElement = get(rootElement)?.element
+	if (!currentRootElement)
+		return console.error('No Root DataTypeTemplates element found')
+
+	const newPrivateElement = pluginStore.addElementToXmlDocument({
+		parentElement: currentRootElement,
+		newElementTagName: 'Private',
+		elementAttributes: {
+			type: 'type-designer-plugin'
+		}
+	})
+
+	privateElement.set({
+		element: newPrivateElement,
+		type: 'type-designer-plugin'
+	})
+}
+
+function createPluginXmlStructure() {
+	if (get(isStructureCreated)) return
+
+	if (!get(rootElement)) createDataTypeTemplateElement()
+	if (!get(privateElement)) createPrivateElement()
+
+	isStructureCreated.set(true)
+}
+
+//==== PUBLIC ACTIONS
+
+function addNewType(columnKey: keyof typeof SCD_ELEMENTS) {
+	createPluginXmlStructure()
+
+	const currentPrivateElement = get(privateElement)?.element
+	if (!currentPrivateElement)
+		return console.error('No Private element found in DataTypeTemplates')
+
+	const iterateType = get(typeElements)[`${columnKey}Types`].length + 1
+
+	const newElementId = `${SCD_ELEMENTS[columnKey].type.baseId}${iterateType}`
+	const newElementName = `${SCD_ELEMENTS[columnKey].type.baseName}${iterateType}`
+
+	const insertElementBefore = SCD_ELEMENTS[columnKey].type.insertBefore
+	const firstElementOfNextType = insertElementBefore
+		? get(typeElements)[insertElementBefore][0]?.element
+		: undefined
+
+	const newElement = pluginStore.addElementToXmlDocument({
```

Comment:
newElement is never used, you can just leave the function execution.

Reply:
done